drv/sidecar-mainboard-controller/src/fan_modules.rs

```diff
@@ -0,0 +1,154 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::{Addr, MainboardController, Reg};
+use bitfield::bitfield;
+use drv_fpga_api::{FpgaError, FpgaUserDesign, WriteOp};
+use userlib::FromPrimitive;
+use zerocopy::{AsBytes, FromBytes};
+
+bitfield! {
+    #[derive(Copy, Clone, PartialEq, Eq, FromPrimitive, AsBytes, FromBytes)]
+    #[repr(C)]
+    pub struct FanModuleStatus(u8);
+    pub enable, set_enable: 0;
```

Comment:
This kinda duplicates information in the `Reg::FAN0_STATE` array of constants.  We _could_ use those constants in the bitfield:
```rust
use Reg::FAN0_STATE;
bitfield! {
    #[derive(Copy, Clone, PartialEq, Eq, FromPrimitive, AsBytes, FromBytes)]
    #[repr(C)]
    pub struct FanModuleStatus(u8);
    pub enable, set_enable: FAN0_STATE::ENABLE.trailing_zeros() as usize;
    pub led, set_led: FAN0_STATE::LED.trailing_zeros() as usize;
    pub present, _: FAN0_STATE::PRESENT.trailing_zeros() as usize;
    pub power_good, _: FAN0_STATE::PG.trailing_zeros() as usize;
    pub power_fault, _: FAN0_STATE::POWER_FAULT.trailing_zeros() as usize;
    pub power_timed_out, _: FAN0_STATE::PG_TIMED_OUT.trailing_zeros() as usize;
}
```
This is probably more robust, but is quite wordy, so I'm not sure.

Reply:
The intention of this bitfield is essentially to do that, so I like tying them more closely together. Good thought.